pkg/sink/kafka/sarama.go

```diff
@@ -122,6 +117,28 @@ func NewSaramaConfig(ctx context.Context, o *Options) (*sarama.Config, error) {
 		return nil, cerror.WrapError(cerror.ErrKafkaInvalidConfig, err)
 	}
 
+	kafkaVersion, err := GetKafkaVersion(ctx, config, o)
+	if err != nil {
+		log.Warn("Can't get Kafka version by broker. ticdc will use default version",
+			zap.String("defaultVersion", kafkaVersion.String()))
+	}
+
+	version, err := sarama.ParseKafkaVersion(o.Version)
+	if err != nil {
+		return nil, cerror.WrapError(cerror.ErrKafkaInvalidVersion, err)
+	}
+	if o.IsAssignedVersion {
+		config.Version = version
+		if version.String() != kafkaVersion.String() {
```

Comment:
according to the api version mapping, if the kafka cluster version is greater than 2.8.0,  `version.String() != kafkaVersion.String()` is always true.